pkg/linter/linter.go

```diff
@@ -157,6 +169,7 @@ var isVarEmptyRegex = regexp.MustCompile("^var/empty/")
 var isCompatPackageRegex = regexp.MustCompile("-compat$")
 var isObjectFileRegex = regexp.MustCompile(`\.(a|so|dylib)(\..*)?`)
 var isSbomPathRegex = regexp.MustCompile("^var/lib/db/sbom/")
+var isDocumentationFileRegex = regexp.MustCompile(`(?:READ(?:ME)?|CREDITS|\.(?:md|docx?|rst|[0-9]))$`)
```

Comment:
I've seen README files named like `READ.ME` before. Maybe just add an optional qualifier for the dot?

I would also add `TODO`

It might be useful to check for .`[0-9][a-zA-Z]` because of e.g. `foo.3p`

Otherwise looks great.